compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -1240,7 +1239,32 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 }
             }
         }
+        // Only if an appropriate error source is not found, check method chain for possible candiates
+        let error_source_not_found = unsatisfied_predicates.is_empty();
+        if error_source_not_found && let Mode::MethodCall = mode && let SelfSource::MethodCall(mut rcvr_expr) = source {
+        while let hir::ExprKind::MethodCall(_path_segment, parent_expr, _args, method_span) =
```

Comment:
I would prefer we called `parent_expr` to something that includes `rcvr`/`receiver` in the name, because the `parent` concept is highly overloaded :)